### ISSUE TITLE
on result is returning undefined

### ISSUE THREAD

**Issue by littlechad** [HBYoon/node-mysql-transaction#5]
Hi

i am getting `undefined` as result when i am doing a `.on('result', function(result) {})`, what did i miss?

```
var trCon = transaction({
  connection: [mysql.createConnection, {
    user: user,
    password: password,
    database: database,
    port: port,
    host: host
  }],
  dynamicConnection: 32,
  idleConnectionCutoffTime: 2000,
  timeout: 1000
});
var chain = trCon.chain();
chain
  .on('commit', function(response) {
    console.log('Comitting...', response);
    callback(null, response);
  })
  .on('rollback', function(err) {
    console.log('Rolling back...', err);
    callback(err);
  });
chain
  .query(cctUpdateQuery)
  .on('result', function(result){
    console.log('Update 1 result: ', result);
  })
  .query(cctllUpdateQuery)
  .on('result', function(result){
    console.log('Update 2 log result: ', result);
  })
  .query(ccUpdateQuery)
  .on('result', function(result){
    console.log('Update 3 result: ', result);
  })
  .query(cclInsertQuery)
  .on('result', function(result){
    console.log('insert 1 result: ', result);
  });
trCon.end();
```

Thank you so much for your time

**Comment by HBYoon** [comment]
I tested code below, and it works :)

```
// DB: mariabd: 10.1.11
// Driver: mysql: 2.10.2
var chain = trCon.chain();
chain
  .on('commit', function(response) {
    // response === undefined
    console.log('Comitting...', response);
  })
  .on('rollback', function(err) {
    console.log('Rolling back...', err);
  });

chain
  .query('insert transaction_test set test=?', [1])
  .on('result', function(result){
    console.log('Insert 1 result: ', result);
  })
  .query('insert transaction_test set test=?', [0])
  .on('result', function(result){
    console.log('Insert 2 log result: ', result);
  });

// why?
trCon.end();
```

by the way, the 'commit' events doesn't have any response data. 

and your code terminate connection object(trCon) in every times. I recommend you reuse the connection object.

cheers

**Comment by littlechad** [comment]
I tested your code, and it is adding to the db perfectly, but it doesn't show any result 

it doesn't event print `console.log('Insert 1 result: '` piece of the code

please help

**Comment by HBYoon** [comment]
umm...in my system(node.js 5.7.0), 'result' event works good.

how are 'commit' and 'rollback' events works?
if these events be good working, at this time, I don't have any solution.

**Comment by littlechad** [comment]
Using this solved my problem, i am on node v5.3.0 btw

``````
trCon.set(function(err, safeCon){
  if (err) {
    return console.log(err);
  }
  safeCon.on('commit', function(){
    console.log('commit');
  });
  safeCon.on('rollback', function(err){
    console.log(err);
  });
  safeCon.query('INSERT INTO testing_transaction(name) VALUES(?)', ['test#1'], function(err,result){
    if (err) {
      safeCon.rollback();
    }
    console.log('Insert 1 result: ', result);
    safeCon.query('INSERT INTO testing_transaction_detail(testing_transaction_id) VALUES(?)', [result.insertId], function(err,result){
      if (err) {
        safeCon.rollback(err);
      }
      console.log('Insert 2 result: ', result);
      safeCon.commit();
    });
  });
});```

thanks man
``````
